dipy/tracking/fbcmeasures.pyx

```diff
@@ -0,0 +1,334 @@
+        r = np.array([1, 1, 0, 0])
+        g = np.array([1, 0, 0, 1])
+        b = np.array([0, 0, 1, 1])
+        fr = interp1d(x, r, bounds_error=False, fill_value=0)
+        fg = interp1d(x, g, bounds_error=False, fill_value=0)
+        fb = interp1d(x, b, bounds_error=False, fill_value=0)
+
+        # select fibers above the RFBC threshold
+        streamlinelist = []
+        rfbclist = []
+        lfbclist = []
+        for i in range(len(self.streamlines_rfbc)):
+            rfbc = self.streamlines_rfbc[i]
+            lfbc = lfbc_log[i]
+            if rfbc > threshold:
+                fiber = np.array(self.streamline_points[i])
+                fiber = fiber[0:self.streamline_length[i] - 1]
+                streamlinelist.append(fiber)
+
+                rfbclist.append(rfbc)
+
+                lfbc = lfbc[0:self.streamline_length[i] - 1]
+                lfbcfiberlist = np.transpose([fr(lfbc), fg(lfbc), fb(lfbc)])
+                lfbclist.append(lfbcfiberlist.tolist())
+
+        return streamlinelist, lfbclist, rfbclist
+    
+    @cython.wraparound(False)
+    @cython.boundscheck(False)
+    @cython.nonecheck(False)
+    @cython.cdivision(True)
+    cdef void compute(self, 
+                      py_streamlines,
+                      kernel,
+                      num_threads=None):
+        """ Compute the fiber to bundle coherence measures for a set of 
+        streamlines.
+
+        Parameters
+        ----------
+        py_streamlines : list
+            A collection of streamlines, each n by 3, with n being the number of
+            nodes in the fiber.
+        kernel : Kernel object
+            A diffusion kernel object created from EnhancementKernel.
+        num_threads : int
+            Number of threads to use for OpenMP.
+        """
+        cdef:
+            int numberOfFibers
+            int maxLength
+            int dim
+            double [:, :, :] streamlines
+            int [:] streamlines_length
+            double [:, :, :] streamlines_tangent
+            int [:, :] streamlines_nearestp
+            double [:, :] streamline_scores
+            double [:] tangent
+            int lineId, pointId
+            int lineId2, pointId2
+            double score
+            int xd, yd, zd
+            double [:, :, :, :, ::1] lut
+            int N
+            int hn
+            int threads_to_use = -1
+            int all_cores = openmp.omp_get_num_procs()
+
+            double [:] score_mp
+            int [:] xd_mp
+            int [:] yd_mp
+            int [:] zd_mp
+
+        if num_threads is not None:
+            threads_to_use = num_threads
+        else:
+            threads_to_use = all_cores
+
+        if have_openmp:
+            openmp.omp_set_dynamic(0)
+            openmp.omp_set_num_threads(threads_to_use)
+        
+        # if the fibers are too short FBC measures cannot be applied, 
+        # remove these.
+        streamlines_length = np.array([len(x) for x in py_streamlines], 
```

Comment:
In case of outliers in fiber length that is true. I will have a look at changing it to an array of arrays implementation. Sorry for delayed reactions -- I am currently attending the ISMRM